### PR TITLE
readme(dm): update dm test readme

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve? 

Suggest IT use MySQL 8.0.21.

### Check List 

Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/tests/README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 1. Run `make dm_integration_test_build` to generate DM related binary for integration test
 
-2. Setup two MySQL servers (the first one: 5.6 ~ 5.7; the second one: 8.0) with [binlog enabled first](https://dev.mysql.com/doc/refman/5.7/en/replication-howto-masterbaseconfig.html) and [set `GTID_MODE=ON`](https://dev.mysql.com/doc/refman/5.7/en/replication-mode-change-online-enable-gtids.html), You need set the mysql port and root password according to the following table.
+2. Setup two MySQL servers (the first one: 5.6 ~ 5.7; the second one: 8.0.21, suggest you are same as [CI](https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/dm/dm_ghpr_new_test.groovy#L164-L172)) with [binlog enabled first](https://dev.mysql.com/doc/refman/5.7/en/replication-howto-masterbaseconfig.html) and [set `GTID_MODE=ON`](https://dev.mysql.com/doc/refman/5.7/en/replication-mode-change-online-enable-gtids.html), You need set the mysql port and root password according to the following table.
```
Comment (lance6716) [inline dm/tests/README.md:36]: that repo is not public

**Review by lance6716 — APPROVED** [review]
did you meet some problems of MySQL version?

**Comment by okJiang** [comment]
> did you meet some problems of MySQL version?

yes. Higher MySQL support [FLUSH_TABLE](https://dev.mysql.com/doc/refman/8.0/en/flush.html#flush-tables) privilege to `flush table with read lock`. So SOME test will failed! e.g. full_mode - -

**Review by Ehco1996 — APPROVED** [review]

**Pull request merged**